src/Microsoft.AspNetCore.SystemWebAdapters.CoreServices/HttpApplication/HttpApplicationOptions.cs

```diff
@@ -0,0 +1,152 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Web;
+
+using static System.FormattableString;
+
+namespace Microsoft.AspNetCore.SystemWebAdapters;
+
+public class HttpApplicationOptions
+{
+    private readonly ModuleCollection _modules = new();
+
+    private Type _applicationType = typeof(HttpApplication);
+
+    public Type ApplicationType
+    {
+        get => _applicationType;
+        set
+        {
+            ArgumentNullException.ThrowIfNull(value);
+
+            _modules.CheckIsReadOnly();
+
+            if (!_applicationType.IsAssignableTo(typeof(HttpApplication)))
+            {
+                throw new InvalidOperationException($"Type {value.FullName} is not a valid HttpApplication");
+            }
+
+            _applicationType = value;
+        }
+    }
+
+    public IDictionary<string, Type> Modules => _modules;
+
+    internal void MakeReadOnly() => _modules.MakeReadOnly();
+
+    /// <summary>
+    /// Gets or sets the number of <see cref="HttpApplication"/> retained for reuse. In order to support modules and appplications that may contain state,
+    /// a unique instance is required for each request. This type should be set to the average number of concurrent requests expected to be seen.
+    /// </summary>
+    public int PoolSize { get; set; } = 100;
```

Comment:
What's the pool size on System.Web?

Reply:
There doesn't appear to be a specified number. They create on demand, then [this method](https://referencesource.microsoft.com/#System.Web/HttpApplicationFactory.cs,c535514db5112a4d,references) is called when there's idle time to trim 3% of the pool on each call

Reply:
This kind of dynamic pool size doesn't seem easy to implement on top of ObjectPool<> as-is, although I did play around with a custom implementation that would use ML.NET and its forecasting APIs to dynamically adapt the target pool size based on the historical concurrent request count :)